eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -230,8 +230,8 @@ object Graph {
           val newMinimumKnownWeight = edgeWeight(edge, currentWeight, initialWeight.length == 0 && neighbor == sourceNode, currentBlockHeight, wr)
 
           // test for ignored edges
-          if (edge.update.htlcMaximumMsat.forall(newMinimumKnownWeight.cost + amountMsat <= _) &&
-            newMinimumKnownWeight.cost + amountMsat >= edge.update.htlcMinimumMsat &&
+          if (edge.update.htlcMaximumMsat.forall(newMinimumKnownWeight.cost <= _) &&
```

Comment:
If `amountMsat` is not used here then it's not used at all in `dijkstraShortestPath` and should be removed from the parameter list. We then need to make sure that https://github.com/ACINQ/eclair/blob/352e4d00cd20c3a1e589d6392c71a1b43298ef48/eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala#L127 is correct

Reply:
With this change `dijkstraShortestPath` will use only the cost found in the `initialWeight` parameter which is `RichWeight(amountMsat, 0, 0, 0)` for the initial step and when calculating the 'spurPath' for an alternative path this is the result of `pathWeight` so it's the cost to go from the starting node to this intermediate node (from here the algorithm will search for a different path). I will remove `amountMsat` from the parameter list.

Reply:
f62f472 removes the unused parameter and also extracts the `initialWeight` parameter in the first invocation of `dijkstraShortestPath`, for more readability.